### PR TITLE
Documentation update: mostly main README, print_array_elements, derive_mirror_rnda; conf.py

### PR DESCRIPTION
**Pull request by GernotMaier**
This PR starts addressing issue #558 to revise and update the documentation. This is a side project and feel free to review whenever there is time.

Main changes:

- removal of docs/sources/index.rst by a file which includes the main repository readme. Both files have been 95% text identical and we have complained about this duplication before. 
- fixed all sphinx warnings
- changed sphinx settings to fail on warnings. We otherwise don't really notice issues in the documentation (e.g., broken links), unless we run sphinx locally or look at the github action output
- removed clutter in sphinx configuration file conf.py . Removed mostly irrelevant lines (e.g., all latex related lines; using latex will anyway require a bit of work and should then be re-introduced). Changed the html-theme to the one used in the CTAO template (which is used by no other project...but it is simple and the fancy one from ctapipe/gammapy is not working for us - this requires work beyond this pull request). **My suggestion is to not spend a lot of time in the review of conf.py, as this will very likely be rebuild in future with a nicer and possibly CTAO conform html theme.**
- simplified and updated the main README to be more in sync with other python projects
- improved description of print_array_elements and derive_mirror_rnda application.

As mentioned, this is part of a revision of all documentation and more changes to follow.

I suggest to build the documentation locally for the review:

```
conda activate simtools-dev
cd docs
make clean
make html
open -a firefox build/index.html
```
(or whatever browser you are using)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -20,41 +29,70 @@ simtools
 .. image:: https://codecov.io/gh/gammasim/simtools/branch/master/graph/badge.svg?token=AYAIRPARCH
    :target: https://codecov.io/gh/gammasim/simtools
 
+Simulation tools and applications for the `Cherenkov Telescope Array (CTAO) <www.cta-observatory.org>`_.
```
Comment (VictorBarbosaMartins) [inline README.rst:32]: This link is not working in the documentation page, I think because we are missing the 'https://'.

File: README.rst [hunk 2]
```diff
@@ -20,41 +29,70 @@ simtools
...
+* I/O and reporting tools for the MC simulation model parameter and production database
 
 simtools is one part of the CTAO Simulation Pipeline, which consist of the following components:
 
-- `CORSIKA <https://www.iap.kit.edu/corsika/>`_ air shower simulation code and the `sim_telarray <https://www.mpi-hd.mpg.de/hfm/~bernlohr/sim_telarray/>`_ telescope simulation code.
-- simtools for tools and application for model parameter and configuration management, database interfaces, validation, and production preparations.
-- databases, especially the model parameter database.
+- `CORSIKA <https://www.iap.kit.edu/corsika/>`_ air shower simulation code and the `sim_telarray <https://www.mpi-hd.mpg.de/hfm/~bernlohr/sim_telarray/>`_ telescope simulation code
+- `workflows <https://github.com/gammasim/workflows>`_ for setting, derivation and validation of simulation model parameters
+- `simulation model parameter and input data schema <https://github.com/gammasim/workflows>`_
```
Comment (VictorBarbosaMartins) [inline README.rst:54]: I'm not sure we want to repeat the same link here. Maybe we can link directly the schema files [(schemas)?](https://github.com/gammasim/workflows/tree/main/schemas)

File: docs/source/conf.py [hunk 3]
```diff
@@ -110,8 +100,7 @@ def get_authors_from_citation_file(file_name):
 # The suffix(es) of source filenames.
 # You can specify multiple suffix as a list of string:
 #
-# source_suffix = ['.rst', '.md']
-source_suffix = ".rst"
+source_suffix = ['.rst']
```
Comment (VictorBarbosaMartins) [inline docs/source/conf.py:103]: I recall the discussions regarding using markdown but I dont recall the conclusion of the discussions. With this change here, I assume we now decide to keep only the rst files, right?

File: README.rst [hunk 4]
```diff
@@ -20,41 +29,70 @@ simtools
...
+- databases, especially the model parameter database
```
Comment (VictorBarbosaMartins) [inline README.rst:55]: databases is the only one here without a link. Maybe we can link this to `simtools/docs/build/html/databases.html#model-parameters-db`.

**Review by VictorBarbosaMartins — APPROVED** [review]
Very nice update. The theme looks indeed much better than the one we had before.

I leave a couple of comments only. Beside these, I have two more general comments:

- I have realized that there are still some commands to run applications in the old style (i.e. `python applications/...`). I assume we want to change that, but maybe not necessarily in this PR.
- I am wondering if we could at some point also include the documentation for the workflows in the main documentation page (index), perhaps before Applications and after Configuration.

File: README.rst (hunk 1 above)
Comment (GernotMaier) [reply README.rst:32]: Fixed.

File: README.rst (hunk 2 above)
Comment (GernotMaier) [reply README.rst:54]: Agree. Fixed.

File: docs/source/conf.py (hunk 3 above)
Comment (GernotMaier) [reply docs/source/conf.py:103]: This is the "preparation" for it. I will add '.md' as soon as we actually add a markdown file.

File: README.rst (hunk 4 above)
Comment (GernotMaier) [reply README.rst:55]: OK, done!

**Pull request merged**